### PR TITLE
[text generation] fix issues with top_p

### PR DESCRIPTION
**Pull request by bfineran**
fixes two issues with `top_p`:

1. setting filtered values to `-inf` is incompatible with `numpy.choice`, was causing issues with numpy.choice in certain scenarios
2. fixes top_p application to grab the true top `p` percent by weight of the logits softmax distribution (previously was sorting  descending causing the least likely tokens would have the highest density)

**minimal reproducible example:**
```python
from deepsparse import Pipeline

pipeline = Pipeline.create(
    "text-generation",
    model_path="hf:neuralmagic/mpt-7b-chat-pruned50-quant",
    engine_type="onnxruntime",
    sequence_length=32,
    prompt_sequence_length=1,
)
generation_config = dict(top_p=0.5, max_new_tokens=5, do_sample=True)
pipeline("def hello_world:", generation_config=generation_config)
```

**test_plan:**
@horheynm to add unit tests in ongoing refactor

### CODE REVIEW

**Review by horheynm — DISMISSED** [review]

**Review by mgoin — DISMISSED** [review]
nice catches, thank you!

**Review by horheynm — APPROVED** [review]

**Review by mgoin — APPROVED** [review]
